internal/version/service.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package version
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/sync2"
+)
+
+// Config contains the necessary Information to check the Software Version
+type Config struct {
+	ServerAddress  string        `help:"server address to check its version against" default:"https://satellite.stefan-benten.de/version"`
+	RequestTimeout time.Duration `help:"Request timeout for version checks" default:"0h1m0s"`
+	CheckInterval  time.Duration `help:"Interval to check the version" default:"0h15m0s"`
+}
+
+// Service contains the information and variables to ensure the Software is up to date
+type Service struct {
+	config  *Config
+	info    *Info
+	service string
+
+	Loop *sync2.Cycle
+
+	checked chan struct{}
+	mu      sync.Mutex
+	allowed bool
+}
+
+// NewService creates a Version Check Client with default configuration
+func NewService(config *Config, info *Info, service string) (client *Service) {
+	return &Service{
+		config:  config,
+		info:    info,
+		service: service,
+		Loop:    sync2.NewCycle(config.CheckInterval),
+		checked: make(chan struct{}, 0),
+		allowed: false,
+	}
+}
+
+// Run logs the current version information
+func (srv *Service) Run(ctx context.Context) error {
+	firstCheck := true
+	return srv.Loop.Run(ctx, func(ctx context.Context) error {
+		var err error
+		allowed, err := srv.checkVersion(ctx)
```

Comment:
gotta check the error here

Reply:
very important that if we have any errors, we let the process keep running

Reply:
reasonable go standards are to assume `allowed` could be anything (probably false) if `err != nil`

Reply:
Check happens in Line 71 and is just a simple error log message